ui/src/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -169,7 +170,54 @@ export function WorkflowLogsViewer({workflow, initialNodeId, initialPodName, con
     // default to the node id of the pod
     const nodeId = initialNodeId || podNamesToNodeIDs.get(podName);
     const node = workflow.status.nodes[nodeId];
-    const templates = execSpec(workflow).templates.filter(t => !node || t.name === getTemplateNameFromNode(node));
+
+    useEffect(() => {
+        if (!node || !node.templateRef) {
+            setTemplateFromRef(null);
+            return;
+        }
+
+        const fetchTemplate = async () => {
+            try {
+                if (node.templateRef.clusterScope) {
+                    console.log(`Fetching template from ClusterWorkflowTemplate: ${node.templateRef.name}, template: ${node.templateRef.template}`);
+                    const tmpl = await services.clusterWorkflowTemplate.get(node.templateRef.name);
```

Comment:
Then, some duplicated code can be removed if you do 
```suggestion
                    const tmpl = node.templateRef.clusterScope ?
                        await services.clusterWorkflowTemplate.get(node.templateRef.name) :
                        await services.workflowTemplate.get(node.templateRef.name, workflow.metadata.namespace)
```